applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -409,73 +437,97 @@ namespace MPMParticleGeneratorUtility
-                                //p_condition->SetValuesOnIntegrationPoints(MPC_CONDITION_ID, mpc_condition_id, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_COORD, mpc_xg , process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_AREA,  mpc_area  , process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_NORMAL, { mpc_normal }, process_info);
+                                // Particle condition are not created twice
+                                bool create_condition = true;
+                                // loop only necessary for equal particle distribution to avoid doubled conditions
+                                if (is_equal_int_volumes){
+                                    for(auto it=ParticleConditions.begin(); it!=ParticleConditions.end(); ++it)
+                                    {
+                                        it->CalculateOnIntegrationPoints(MPC_COORD, xg_tmp, rMPMModelPart.GetProcessInfo());
+                                        
+                                        if (xg[0] == xg_tmp[0][0] && xg[1] == xg_tmp[0][1]  && xg[2] == xg_tmp[0][2] )
+                                        {
+                                            create_condition = false;
+                                            it->CalculateOnIntegrationPoints(MPC_AREA, area_temp, rMPMModelPart.GetProcessInfo());
+                                            area_temp[0] += mpc_area[0];
+                                            it->SetValuesOnIntegrationPoints(MPC_AREA, area_temp, rMPMModelPart.GetProcessInfo());
+                                            
+                                        }
+                                    }
+                                }
 
+                                if (create_condition){
 
-                                p_condition->SetValuesOnIntegrationPoints(MPC_DISPLACEMENT, { mpc_displacement }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_DISPLACEMENT, { mpc_imposed_displacement }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_VELOCITY, { mpc_velocity }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_VELOCITY, { mpc_imposed_velocity }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_ACCELERATION, { mpc_acceleration }, process_info);
-                                p_condition->SetValuesOnIntegrationPoints(MPC_IMPOSED_ACCELERATION, { mpc_imposed_acceleration }, process_info);
-                                // Mark as boundary condition
-                                p_condition->Set(BOUNDARY, true);
+                                    // Create new material point condition
+                                    new_condition_id = last_condition_id + 1 ;
```

Comment:
Can we use only `last_condition_id` or we also need `new_condition_id`? Because `last_condition_id` is updated in the same way at the end of this `if` condition

Reply:
I'll remove one of those variables